WIP 0.3.0

crisptrutski opened the pull request:
@Deraen Let's use this as an integration branch to PR against for any features/fixes to include in the next breaking release.

I you have the patience (or already did the work), would also take same contributions against `master` and would cut PATCH releases for them. 

But would prefer not to port extra stuff over now, so favour here :smile:

Deraen commented:
I can rework my changes on top of this. I guess the some of the changes I did are already fixed by this. I'll try to find time to check this on weekend.

If this is ready, I think it would be easiest to just merge this to master and I'll make my PRs against that.

crisptrutski commented:
This could still use some polish even on what I've covered already (for example printing warnings for removed arguments), so will leave it open for bit.

Thanks for chipping in and for all the guidance so far - if I have time may port the work you've already shared, and implement what I can of the other issues you've opened.

crisptrutski commented on src/crisptrutski/boot_cljs_test.clj:
```diff
@@ -27,145 +23,171 @@
+        edn        (when edn? (read-string (slurp (core/tmp-file (core/tmp-get fileset edn-path)))))
+        ;; subset namespaces to those directly required by .cljs.edn
+        ;; TODO: better would be to use transitive requires, once we have the dep graph
+        namespaces (if edn (filter (set (:require edn)) namespaces) namespaces)
```
@Deraen one part I'm not super happy about, but better than errors trying to run a namespace that hasn't loaded

crisptrutski commented on src/crisptrutski/boot_cljs_test.clj:
```diff
@@ -27,145 +23,171 @@
+      (if suite?
+        (info "Using %s...\n" edn-path)
+        (do (info "Updating %s...\n" edn-path)
+            (spit edn-file (update edn :require (fn [xs] (when-not (some #{suite-ns} xs)
```
A similar wart, let me know if you have any immediate thoughts on these niggles

Commit:
Flattened 0.3.0 work

Commit:
Thin out plans for version

Commit:
Whitespace. Forget about transitive deps for edn files - transitive test namespace requires seem odd.

Commit:
Meh bleh

Commit:
Whitespace etc

Commit:
Catch missing doo require

Commit:
Fixup missing arg

Commit:
Use recent cljs version

Commit:
Switch to doo release version

Commit:
More pedantic default name

Commit:
Use exceptions instead of System/exit

Commit:
... missed

Commit:
Rename

Commit:
Error smuggling util upgrade

Commit:
Add some noise

Commit:
..track rename missing

Commit:
Add warning sounds

Commit:
WIP karma support

Need to sync `node_modules` in a better way to avoid single-segment namespace warning

Commit:
Match boot's 1 based, not 0 based verbosity

Commit:
Using changelog as todo list

Commit:
Basic Karma support

Commit:
Longer wait needed

Commit:
Remove misleading option description

Commit:
Convert filename to OS style

Commit:
Nil guard for file-not-found

Commit:
More 'keep him alive for the execution' guards

Commit:
Fixup windows support

Commit:
Boot 2.6.0

Commit:
Update docs and improve plumbing api

Commit:
Update Changlog. Adjust verbosity scale.

Commit:
README tweaks

crisptrutski committed:
A few more README tweaks

crisptrutski committed:
Tweaking out

crisptrutski committed:
Typos
